docs/reference-guide/storage-modes.md

```diff
@@ -117,4 +146,119 @@ In the `IN_MEMORY_ANALYTICAL` mode, periodic snapshots are **disabled**.
 query is run in the `IN_MEMORY_ANALYTICAL` mode, Memgraph guarantees that it
 will be **the only** transaction present in the system, and all the other
-transactions will wait until the snapshot is created to ensure its validity.
\ No newline at end of file
+transactions will wait until the snapshot is created to ensure its validity.
+
+## On-disk transactional storage mode
+
+In the on-disk transactional storage mode, disk is used as a physical storage
+which allows you to save more data than the capacity of your RAM. This helps
+keep the hardware costs to a minimum, but you should except slower performance
+when compared to the in-memory transactional storage mode. Keep in mind that
+while executing queries, all the graph objects used in the transactions still
+need to be able to fit in the RAM, or Memgraph will throw an exception. 
+
+### Architecture
+
+RocksDB is used as a background storage to serialize nodes and relationships
+into a key-value format. The used architecture is also known as "larger than
+memory" as it enables in-memory databases to save more data than the main memory
+can hold, without the performance overhead caused by the buffer pool. 
+
+The imported data is residing on the disk, while the main memory contains two
+caches, one executing operations on main RocksDB instance and the other for
+operations that require indices. The cache used here is Memgraph's custom
+`SkipList` cache which allows multithreaded read-write access pattern.
+
+### MVCC
+
+Concurrent execution of transactions is supported differently for on-disk
+storage than for in-memory. The in-memory storage mode relies on delta objects
+which store the exact versions of data at the specific moment in time.
+Therefore, the in-memory storage mode uses a pessimistic approach and
+immediately checks whether there is a conflict between two transactions. 
+
+In the on-disk storage mode, the cache is used per transaction. This
+significantly simplifies object management since there is no need to question
+certain object's validity, but it also requires the optimistic approach for
+conflict resolution between transactions. 
+
+In the on-disk storage mode, the conflict is checked at the transaction's commit
+time with the help of RocksDB's transaction support. This also implies that
+deltas are cleared after each transaction, which can optimize memory usage
+during execution. Deltas are still used to fully support Cypher's semantic of
+the write queries. The design of the on-disk storage also simplifies the process
+of garbage collection, since all the data is on disk.
+
+### Isolation level
+
+The on-disk storage mode supports only snapshot isolation level. Mostly because
+it's the Memgraph viewpoint that snapshot isolation should be the default
+isolation level for most applications relying on databases. But the snapshot
+isolation level also simplifies the query's execution flow since there is no
+need to go on disk until the commit of the transaction.
```

Comment:
```suggestion
isolation level also simplifies the query's execution flow since no data is transferred to the disk until the commit of the transaction.
```
 ok like this?

Reply:
yes